src/components/checkbox/checkbox.ts

```diff
@@ -0,0 +1,246 @@
+   * be interacted with. The correct ARIA attributes are applied to denote this to assistive
+   * technology.
+   */
+  @Input() disabled: boolean = false;
+
+  /**
+   * The tabindex attribute for the checkbox. Note that when the checkbox is disabled, the attribute
+   * on the host element will be set to -1, regardless of the actual tabindex value.
+   */
+  @Input() tabindex: number = 0;
+
+  /** Event emitted when the checkbox's `checked` value changes. */
+  @Output() change: EventEmitter<boolean> = new EventEmitter();
+
+  /** Called when the checkbox is blurred. Needed to properly implement ControlValueAccessor. */
+  onTouched: () => any = () => {};
+
+  private _currentAnimationClass: string = '';
+
+  private _currentUiState: UiTransitionState = UiTransitionState.Init;
+
+  private _checked: boolean = false;
+
+  private _indeterminate: boolean = false;
+
+  private _changeSubscription: {unsubscribe: () => any} = null;
+
+  constructor(private _renderer: Renderer, private _elementRef: ElementRef) {}
+
+  /**
+   * True when the checkbox is checked, false otherwise. Note that setting `checked` while
+   * will immediately set `indeterminate` to false.
+   */
+  @Input() get checked() {
+    return this._checked;
+  }
+
+  set checked(checked: boolean) {
+    // When checked is set, it immediately overwrites an indeterminate state, regardless of
+    // whether it's true or false.
+    this._indeterminate = false;
+    this._checked = checked;
+    this._transitionUiState(
+        this._checked ? UiTransitionState.Checked : UiTransitionState.Unchecked);
+    this.change.emit(this._checked);
+  }
+
+  /**
+   * True when the checkbox is indeterminate, false otherwise. This is also known as "mixed" mode
+   * and can be used to represent a checkbox with three states, e.g. a checkbox that represents a
+   * nested list of checkable items.
+   */
+  @Input() get indeterminate() {
+    return this._indeterminate;
+  }
+
+  set indeterminate(indeterminate: boolean) {
+    this._indeterminate = indeterminate;
+    if (this._indeterminate) {
+      this._transitionUiState(UiTransitionState.Indeterminate);
+    } else {
+      this._transitionUiState(
+          this.checked ? UiTransitionState.Checked : UiTransitionState.Unchecked);
+    }
+  }
+
+  /** The id that is attached to the checkbox's label. */
+  get labelId() {
+    return `${this.id}-label`;
+  }
+
+  /** Returns the proper aria-checked attribute value based on the checkbox's state. */
+  getAriaChecked() {
+    if (this.indeterminate) {
+      return 'mixed';
+    }
+    return this.checked ? 'true' : 'false';
+  }
+
+  /** Toggles the checked state of the checkbox. If the checkbox is disabled, this does nothing. */
+  toggle(event: Event) {
+    if (this.disabled) {
+      event.stopPropagation();
+      return;
+    }
+    this.checked = !this.checked;
+  }
+
+  /** Implemented as part of ControlValueAccessor. */
+  writeValue(value: any) {
+    this.checked = !!value;
```

Comment:
Dartpad doesn't use checked mode.

https://dartpad.dartlang.org/8f79d250451143ec113c shows my results when run locally in checked mode.